Mathlib/Data/Sym/Card.lean

```diff
@@ -182,31 +184,18 @@ theorem card_subtype_diag [Fintype α] : card { a : Sym2 α // a.IsDiag } = card
 theorem card_subtype_not_diag [Fintype α] :
     card { a : Sym2 α // ¬a.IsDiag } = (card α).choose 2 := by
   convert card_image_offDiag (univ : Finset α)
-  simp_rw [Quotient.mk', ← Quotient.mk''_eq_mk] -- Porting note: Added `simp_rw`
-  rw [Fintype.card_of_subtype, ← filter_image_quotient_mk''_not_isDiag]
+  -- Porting note (kmill): Quotient.mk and Quotient.mk'' are syntactically different.
+  -- `filter_image_quotient_mk''_not_isDiag` was part of rw
+  refine Eq.trans ?_ congr(Finset.card $(filter_image_quotient_mk''_not_isDiag univ))
+  rw [Fintype.card_of_subtype]
   rintro x
   rw [mem_filter, univ_product_univ, mem_image]
   obtain ⟨a, ha⟩ := Quotient.exists_rep x
   exact and_iff_right ⟨a, mem_univ _, ha⟩
 #align sym2.card_subtype_not_diag Sym2.card_subtype_not_diag
 
-/-- Finset **stars and bars** for the case `n = 2`. -/
-theorem _root_.Finset.card_sym2 (s : Finset α) : s.sym2.card = s.card * (s.card + 1) / 2 := by
-  rw [← image_diag_union_image_offDiag, card_union_eq, Sym2.card_image_diag,
-    Sym2.card_image_offDiag, Nat.choose_two_right, add_comm, ← Nat.triangle_succ, Nat.succ_sub_one,
-    mul_comm]
-  rw [disjoint_left]
-  rintro m ha hb
-  rw [mem_image] at ha hb
-  obtain ⟨⟨a, ha, rfl⟩, ⟨b, hb, hab⟩⟩ := ha, hb
-  refine' not_isDiag_mk'_of_mem_offDiag hb _
-  dsimp [Quotient.mk'] at hab -- Porting note: Added `dsimp`
-  rw [hab]
-  exact isDiag_mk'_of_mem_diag ha
-#align finset.card_sym2 Finset.card_sym2
-
 /-- Type **stars and bars** for the case `n = 2`. -/
-protected theorem card [Fintype α] : card (Sym2 α) = card α * (card α + 1) / 2 :=
+protected theorem card [Fintype α] : card (Sym2 α) = Nat.choose (card α + 1) 2 :=
   Finset.card_sym2 _
 #align sym2.card Sym2.card
```

Comment:
I can't find where the `Fintype` instance is anymore, but shouldn't this be moved to `Data.Finset.Sym`, or maybe `Data.Fintype.Sym` if we really really care about import scope creep to the point of excluding `Fintype` from those files (and I think we don't care).

Reply:
The Fintype instance is added in this PR, `instFintypeSym2`, in `Data.Finset.Sym`. There's already enough Fintype stuff imported in that module that it can be defined right there, but someone could refactor and the things that depend on it out if it's worth doing.